libs/data_providers_sdk/src/price_data/fetchers/exchanges/gemini.rs

```diff
@@ -33,20 +33,25 @@ impl<'a> PricesFetcher<'a> for GeminiPriceFetcher<'a> {
         Self { symbols }
     }
 
-    fn fetch(&self) -> LocalBoxFuture<Result<PairPriceData>> {
-        async {
+    fn fetch(&self, timeout_secs: u64) -> LocalBoxFuture<Result<PairPriceData>> {
+        async move {
             let prices_futures = self
                 .symbols
                 .iter()
                 .map(Deref::deref)
-                .map(fetch_price_for_symbol);
+                .map(|symbol| fetch_price_for_symbol(symbol, timeout_secs));
 
             let mut futures = FuturesUnordered::from_iter(prices_futures);
             let mut prices = PairPriceData::new();
 
             while let Some(result) = futures.next().await {
-                if let Ok((symbol, price_pint)) = result {
-                    prices.insert(symbol, price_pint);
+                match result {
+                    Ok((symbol, price_pint)) => {
+                        prices.insert(symbol, price_pint);
+                    }
+                    Err(err) => {
+                        anyhow::bail!("Error processing future in GeminiPriceFetcher {err:?}")
```

Comment:
Use `eprintln!` or another way to log. Do not bail, as if we have 10 assets = 10 requests, and one of them fails, but the rest succeed, we do not want to nuke all.